packages/emitter-framework/test/typescript/components/value-expression.test.tsx

```diff
@@ -0,0 +1,236 @@
+  );
+});
+
+it("renders empty objects", async () => {
+  // Can be replaced with with TypeKit once #6976 is implemented
+  const program = await getProgram(`
+      namespace DemoService;
+      @example(#{})
+      model ObjectValue {};
+    `);
+  const [namespace] = program.resolveTypeReference("DemoService");
+  const model = (namespace as Namespace).models.get("ObjectValue");
+  assert.exists(model, "unable to find ObjectValue model");
+
+  const value = getExampleValue(model);
+  await testValueExpression(value, `{}`);
+});
+
+it("renders objects with properties", async () => {
+  // Can be replaced with with TypeKit once #6976 is implemented
+  const program = await getProgram(`
+      namespace DemoService;
+      @example(#{a: 5, b: "foo", c: true})
+      model ObjectValue {
+        a: int32;
+        b: string;
+        c: boolean;
+      };
+    `);
+  const [namespace] = program.resolveTypeReference("DemoService");
+  const model = (namespace as Namespace).models.get("ObjectValue");
+  assert.exists(model, "unable to find ObjectValue model");
+
+  const value = getExampleValue(model);
+  await testValueExpression(
+    value,
+    dedent(`
+      {
+        a: 5,
+        b: "foo",
+        c: true,
+      }`),
+  );
+});
+
+it("renders enums", async () => {
+  // Can be replaced with with TypeKit once #6976 is implemented
+  const program = await getProgram(`
+      namespace DemoService;
+      enum Color {
+        Red,
+        Green: 3,
+        Blue
+      }
+    `);
+  const [namespace] = program.resolveTypeReference("DemoService");
+  const colors = (namespace as Namespace).enums.get("Color");
+  assert.exists(colors, "unable to find Color enum");
+
+  const red = colors?.members.get("Red");
+  assert.exists(red, "unable to find Red enum member");
+  await testValueExpression(
+    {
+      valueKind: "EnumValue",
+      value: red,
+    } as EnumValue,
+    `"Red"`,
+  );
+
+  const green = colors?.members.get("Green");
+  assert.exists(green, "unable to find Green enum member");
+  await testValueExpression(
+    {
+      valueKind: "EnumValue",
+      value: green,
+    } as EnumValue,
+    `3`,
+  );
+});
+
+/**
+ * Helper that renders a value expression and checks the output against the expected value.
+ */
+async function testValueExpression(value: Value, expected: string) {
+  const prefix = "const val = ";
+  const res = render(
+    <Output>
+      <SourceFile path="test.ts">
+        {prefix}
+        <ValueExpression value={value} />
+      </SourceFile>
+    </Output>,
+  );
+  const testFile = res.contents.find((file) => file.path === "test.ts");
+
+  assert.exists(testFile, "test.ts file not rendered");
+
+  assert.equal(
+    testFile.contents,
+    `${prefix}${expected}`,
+    "test.ts file contents do not match expected",
+  );
+}
+
+/**
+ * Extracts the value marked with the @example decorator from a model.
+ */
+function getExampleValue(model: Model): Value {
```

Comment:
I was more thinking that this would take the value string as an input and just abstract teh rest as well(resolve the reference if needed)

Reply:
I think as a simpler way you do do 
```ts
const {prop} = compile(`model Foo {
  @test prop: unknown = ${value}
}`
```

and retrieve `defaultValue` on the property

Reply:
Yeah I figured 😄 but I prefer this as it allows me to name my models something that makes sense instead of the same generic name. I don't like going too much into helpers and utils in tests as it reduces readability and you end up just adding a bunch of functions anyway (getExampleValue, getValueFromModel, getScalar, compileprogram, compileProgramAndGetValue, etc) and just end up with too much indirection.

Since I'll be moving to TypeKits for most of this anyway I'd prefer what I have here